### ISSUE TITLE
Import errors with numpy and python 3.8+

### ISSUE THREAD

**Issue by jacopok** [jacopok/mlgw_bns#1]
The traceback is:

```
poetry run tox -e py38
...
  • Installing numpy (1.21.5): Skipped for the following reason: Already installed
...
py38 run-test: commands[6] | poetry run pytest -v --cov=mlgw_bns --junit-xml tests/test_results.xml
RuntimeError: module compiled against API version 0xf but this version of numpy is 0xe
ImportError while loading conftest '/home/jacopo/Documents/masters/mlgw_bns/tests/conftest.py'.
tests/conftest.py:9: in <module>
    from EOBRun_module import EOBRunPy  # type: ignore
E   ImportError: numpy.core.multiarray failed to import
ERROR: InvocationError for command /home/jacopo/Documents/masters/mlgw_bns/.tox/py38/bin/poetry run pytest -v --cov=mlgw_bns --junit-xml tests/test_results.xml (exited with code 4)
_____________________________________________________________________________________________________ summary _____________________________________________________________________________________________________
ERROR:   py38: commands failed
```

For now, removing py38 from the test suite.

**Comment by jacopok** [comment]
It seems like upgrading numpy to version 1.22 would solve the problem, however that is not compatible with numba; the preferred approach forward would be to find a way to not have issues with numpy 1.21. Is some part of the code assuming we are on 1.22?

**Comment by jacopok** [comment]
The correspondence between the C API versions and numpy versions can be found at 
https://github.com/numpy/numpy/blob/main/numpy/core/setup_common.py#L35

Specifically, the module (`EOBRun_Module`?) was compiled with 0xf -> 1.22+ but it is trying to be run with 1.21.

Where is this numpy 1.22 installation in the compilation of EOBRun? numpy should always be at 1.21 within poetry, since that's what is in the lock.

**Comment by jacopok** [comment]
After regenerating the `.tox` folder the tests run locally for py37 and py39, while in the CI pipeline py37 works, while py39 does not. Not sure what to make of it -- in the CI pipeline everything should be brand new, so I don't see where a 1.22+ version of numpy could have been fished out of!

**Comment by jacopok** [comment]
Solved by #34.
